tests/platform/fw_utility/fwutil_helper.py

```diff
@@ -0,0 +1,617 @@
+                sys.path.append(sub_folder_dir)
+            from check_hw_mgmt_service import check_hw_management_service
+            from check_sysfs import check_sysfs
+
+            logging.info("Check the hw-management service")
+            check_hw_management_service(dut)
+
+            logging.info("Check sysfs")
+            check_sysfs(dut)
+
+
+    def check_version(self, version_to_install, comp_data):
+        """
+        check if bios version was updated as expected.
+        """
+        if comp_data['version'].startswith(version_to_install):
+            return SUCCESS_CODE
+        return FAILURE_CODE
+
+    def get_component_name(self):
+        return 'BIOS'
+
+
+class CpldComponent(FwComponent):
+
+    def parse_version(self, files_path, file_name, fw_data):
+        fw_path = os.path.join(files_path, file_name)
+        real_path = os.path.realpath(fw_path)
+        rev = os.path.basename(real_path).upper().split('REV')
+        revisions = []
+        counts = {}
+        is_latest = False
+        for i in range(len(rev)-1):
+            r = rev[i+1]
+            if r.startswith('_'):
+                r = r.split('_')[1]
+                revisions.append(int(r[0:2]))
+            elif '_' in r:
+                r = r.split('_')[0]
+                revisions.append(int(r[0:2]))
+            else:
+                r = r.split('.')[0]
+                revisions.append(int(r[0:2]))
+
+        for r in revisions:
+            counts[r] = revisions.count(r)
+
+        current_counts = {}
+        current_ver = fw_data['CPLD']['version'].split('.')
+        for r in current_ver:
+            current_counts[int(r)] = current_ver.count(r)
+
+        if counts == current_counts:
+            is_latest = True
+
+        # TODO: for now returning only version and that is not latest
+        return fw_data['CPLD']['version'], False
+        # return counts, is_latest
+
+
+    def get_version(self, dut, files_path, fw_data):
+        # currently taken from revisions but without known order
+        files_path = os.path.join(files_path, 'cpld')
+        versions = {}
+        platform_type = dut.facts['platform']
+        is_latest = False
+        latest_ver = ''
+        other_ver = ''
+        latest = platform_type + '_latest'
+        latest_fw_path = ""
+        other = platform_type + '_other'
+        other_fw_path = ""
+
+        for file_name in os.listdir(files_path):
+            if file_name.startswith(latest):
+                latest_ver, is_latest = self.parse_version(files_path, file_name, fw_data)
+                latest_fw_path = os.path.join(files_path, file_name)
+            if file_name.startswith(other):
+                other_ver, is_other = self.parse_version(files_path, file_name, fw_data)
+                other_fw_path = os.path.join(files_path, file_name)
+
+        versions = {
+            'latest_version': latest_ver,
+            'latest_path': latest_fw_path,
```

Comment:
@noaOrMlnx Looks like we have mismatch here since we are using current FW version in system + latest FW path